byzcoin/bcadmin/test.sh

```diff
@@ -61,8 +62,8 @@ testReset(){
   rm -f config/* *.db
   runCoBG 1 2 3
   runBA create public.toml --interval .5s
-  bc=config/bc*cfg
-  key=config/key*cfg
+  bc=$( echo config/bc*cfg )
+  key=$( echo config/key*cfg )
```

Comment:
I was also surprised to find that
```bash
touch one.tv
a=*.tv
echo $a
touch two.tv
echo $a
```

Outputs

```
one.tv
one.tv two.tv
```

So while in all the tests it just worked, it was kind of random, because there were no other `key*.cfg`. So at least for the user one I needed to have the first key, else the `rm $key` will remove _both_ keys.

But I can remove it from the tests where it's not really needed and more reflects what I just learnt about bash.

Reply:
I agree, shell expension can be weird: for example, where is no match, it will keep the string as it:
```sh
a=no_match*
echo $a # -> no_match*
```

In your case, `a` equals `*.tv` and gets expanded when used, you can avoid the expansion by double-quoting it. I usually avoid as much as possible globing, or I `cut` out the one I want.

Reply:
For the `testUser`, the `key=key*.cfg` starts with the key of the setup. Once the `user connect` is done, there are two keys: an old one for the setup, and a new one for the user. To make sure that the one from the user is valid, I delete the old one. But doing `rm $key` doesn't work, as it expands to _both_ keys. This is why in `testUser`, I do need `key=${ echo key*.cfg`, to really only have the setup key.

But for the other tests it doesn't make a difference...